neuralmagic/benchmarks/run_benchmark_serving.py

```diff
@@ -0,0 +1,71 @@
+import subprocess
+import time
+import socket
+
+from typing import NamedTuple
+from pathlib import Path
+
+from neuralmagic.tools.call_cmd import call_cmd
+from common import download_datasets, script_args_to_cla
+
+BENCH_SERVER_HOST = "localhost"
+BENCH_SERVER_PORT = 9000
+
+def get_this_script_dir() -> Path:
+    return Path(__file__).parent.resolve()
+
+def is_server_running(host: str, port: int, timeout=20) -> bool:
+    def try_connection() -> bool:
+        try:
+            sock = socket.create_connection((host, port))
+            sock.close()
+            return True
+        except Exception as _:
+            return False
+
+    retries = 5
+    timeout_part = timeout / retries
+    while retries:
+        time.sleep(timeout_part)
+        if try_connection():
+            return True
+        retries = retries - 1
+
+    return False
+
+def run_benchmark_serving_script(config: NamedTuple, output_directory: Path) -> None:
+
+    def run_bench(server_cmd: str, bench_cmd: list[str]) -> None:
```

Comment:
soliciting opinion : This function, 
 -  starts the server
 - runs the benchmark 
 - kills the server
 
 However this doesn't represent the nominal state of the engine. I think it should be something like, 
 - start the server
 - Pass in some warmup queries. 
 - run the benchmark 
 - kill the server
 What do you guys think ? 
@rib-2  @mgoin @tlrmchlsmth

Reply:
I'm generally in favor of adding some warmup time to the benchmarks to eliminate any time due to initialization or downclocking. Would we need to modify the API to let us do that?

Reply:
I was asking about warmups from the perspective of the engine state. `starting with a fresh engine` vs `using an engine that has been running for a while`. I think the second is representative of people will use - the first is what we are benchmarking now.

Reply:
Agreed - probably best to use some random data for this if you can (to avoid things where the prefix caching helps us)

Reply:
Added a warmup step for both the serving and static-engine usecase 👍